query.go

```diff
@@ -271,6 +271,11 @@ func (itr *queryIter) Next(out interface{}) bool {
 		return false
 	}
 
+	// stop if exceed limit
+	if itr.output != nil && int(*itr.input.Limit) == itr.idx {
```

Comment:
Thanks! Please put a `&& itr.input.Limit != nil` before `*itr.input.Limit` part. I think it might break on queries without limits.